extensions/raindrop-io/src/hooks/useBrowserLink.ts

```diff
@@ -0,0 +1,27 @@
+import { Toast, getFrontmostApplication, showToast } from "@raycast/api";
+import { runAppleScript, usePromise } from "@raycast/utils";
+
+export function useBrowserLink() {
+  return usePromise(
+    async () => {
+      const app = await getFrontmostApplication();
+
+      switch (app.bundleId) {
+        case "company.thebrowser.Browser":
+          return runAppleScript(`tell application "Arc" to return URL of active tab of front window`);
+        case "com.google.Chrome":
+          return runAppleScript(`tell application "Google Chrome" to return URL of active tab of front window`);
+        case "com.apple.Safari":
+          return runAppleScript(`tell application "Safari" to return URL of front document`);
```

Comment:
Nice.

Can you add also Firefox?
I tried with this applescript and worked fine:
```
tell application "System Events" to get value of UI element 1 of combo box 1 of toolbar "Navigation" of first group of front window of application process "Firefox"
```

The bundleId seems to be `org.mozilla.firefox`:
```
<key>CFBundleIdentifier</key>
<string>org.mozilla.firefox</string>
```

Reply:
@lardissone 
Thanks for the suggestion!
I wanna add support for Firefox too.
I tried adding the code, but got an error.
I'll check it out, but any clue what might be causing it?

```diff
diff --git a/src/hooks/useBrowserLink.ts b/src/hooks/useBrowserLink.ts
index b15f984..cedea6b 100644
--- a/src/hooks/useBrowserLink.ts
+++ b/src/hooks/useBrowserLink.ts
@@ -13,6 +13,10 @@ export function useBrowserLink() {
           return runAppleScript(`tell application "Google Chrome" to return URL of active tab of front window`);
         case "com.apple.Safari":
           return runAppleScript(`tell application "Safari" to return URL of front document`);
+        case "org.mozilla.firefox":
+          return runAppleScript(
+            `tell application "System Events" to get value of UI element 1 of combo box 1 of toolbar "Navigation" of first group of front window of application process "Firefox"`,
+          );
         default:
           throw new Error(`Unsupported App: ${app.name}`);
       }
```

![raindrop-io 2024-05-21 at 00 06 25](https://github.com/raycast/extensions/assets/1296540/652d4c35-a18a-48d1-861f-4b908725377d)

![CleanShot 2024-05-21 at 00 13 06@2x](https://github.com/raycast/extensions/assets/1296540/60551c6e-b1a4-4646-9084-e4668b188f59)

Reply:
Weird, does it work if you call it from the terminal?

`osascript -e  'tell application "System Events" to get value of UI element 1 of combo box 1 of toolbar "Navigation" of first group of front window of application process "Firefox"'`:

![20240520_121847@2x](https://github.com/raycast/extensions/assets/59826/ddc34dfd-d9cb-4f84-a209-c16128d79f69)


I got the applescript from [this](https://apple.stackexchange.com/a/424375) SO post, maybe it requires to have voiceover disabled in Firefox? I have this value in `0` not `-1` as the post suggests.
![20240520_121809@2x](https://github.com/raycast/extensions/assets/59826/8f0e7f88-e874-49c9-b60f-32f8aab18626)

Reply:
When experimenting with rewriting AppleScript to JXA, I don't know why, but after running `﻿SystemEvents.processes.byName("Firefox").properties()`, you get the URL of the current tab without an error.

I modified the AppleScript as follows. So far, this code seems to work well.

```AppleScript
tell application "System Events"
  set firefox to application process "Firefox"

  -- HACK:
  -- If you don't get the properties first, you'll get the error:
  -- `Can't get group 1 of window 1 of application process "Firefox". Invalid index. (-1719)`.
  get properties of firefox

  set frontWindow to front window of firefox
  set firstGroup to first group of frontWindow
  set navigation to toolbar "Navigation" of firstGroup
  get value of UI element 1 of combo box 1 of navigation
end tell
```

Reply:
@lardissone 
I've added a commit to the pull request, please check it out 😄

Reply:
Good catch! Thanks!